File: ios-objc/tooling/RIBs.xctemplate/Default/___FILEBASENAME___Router.m
//___FILEHEADER___

#import "___FILEBASENAME___.h"

@interface ___VARIABLE_productName___Router ()

@property (nonatomic, weak) id<___VARIABLE_productName___ViewControllable> viewController;

@end

@implementation ___VARIABLE_productName___Router
- (instancetype)initWithInteractor:(id<___VARIABLE_productName___Interactable>)interactor viewController:(id<___VARIABLE_productName___ViewControllable>)viewController childBuilders:(nullable NSDictionary<NSString *, id<BRKBuildable>> *)children {
    self = [super initWithInteractor:interactor childBuilders:children];
    if (self) {
        _viewController = viewController;
        interactor.router = self;
    }
    return self;
}


@end

File: ios/tooling/RIB.xctemplate/ownsView/___FILEBASENAME___Router.swift
//___FILEHEADER___

import RIBs

protocol ___VARIABLE_productName___Interactable: Interactable {
    var router: ___VARIABLE_productName___Routing? { get set }
    var listener: ___VARIABLE_productName___Listener? { get set }
}

protocol ___VARIABLE_productName___ViewControllable: ViewControllable {
    // TODO: Declare methods the router invokes to manipulate the view hierarchy.
}

final class ___VARIABLE_productName___Router: ViewableRouter<___VARIABLE_productName___Interactable, ___VARIABLE_productName___ViewControllable>, ___VARIABLE_productName___Routing {

    // TODO: Constructor inject child builder protocols to allow building children.
    override init(interactor: ___VARIABLE_productName___Interactable, viewController: ___VARIABLE_productName___ViewControllable) {
        super.init(interactor: interactor, viewController: viewController)
        interactor.router = self
    }
}
